[Docs] Clarify how ingester replication happens (#4101)
grafana/mimir: docs/sources/mimir/operators-guide/architecture/components/ingester.md

In the event of a single ingester failure, no time series samples are lost.
If multiple ingesters fail, time series might be lost if the failure affects all the ingesters holding the replicas of a specific time series.

> Note that replication only happens at write time, if an ingester is unavailable for a period where writes are still being written to other ingesters that ingester will never recover those missed samples.

Reviewer: Thanks for working on this! I'm wondering if we can rephrase it more clearly. I don't think I did a great job, so asking help from @osg-grafana 

```suggestion
Note that replication only happens at write time. When an ingester gets back online after an outage, it will not restore the missed samples from other ingesters, but will start ingesting new samples.
```

Reviewer 2: ```suggestion
> **Note:** Replication only happens at write time. If an ingester is unavailable during a period when writes are actively being written to other ingesters, that particular ingester will never recover those missed samples.
```
